index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
```

Comment:
@davidchambers would "longest substrings" be technically correct?

> Takes a regex and a string and returns the **longest** substrings that do not match the regex.

Which, after rereading [what `String#split` does](http://www.ecma-international.org/ecma-262/7.0/index.html#sec-regexp.prototype-@@split), actually not exactly the full description:

> If the regular expression contains capturing parentheses, then each time *separator* is matched the results (including any undefined results) of the capturing parentheses are spliced into the output array.

i.e. this would be surprising to me:

```js
> S.splitOnRegex(/(b|c)/, 'abcd')
['a', 'b', '', 'c', 'd']
```